prov/util/src/util_ep.c

```diff
@@ -76,8 +76,9 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
 	if (!info || !info->ep_attr || !info->rx_attr || !info->tx_attr)
 		return -FI_EINVAL;
 
-	ret = ofi_check_info(util_prov,
-			     util_domain->fabric->fabric_fid.api_version, info);
+	ret = ofi_prov_check_info(util_prov,
+				  util_domain->fabric->fabric_fid.api_version,
+				  info, NULL);
```

Comment:
Exactly the problem was here, because we checked only first info item from provider's info list and it was not matched with passed info by the app. Due to this fact in current implementation we go through the entire provider's list to found matching info.